File: hms-frontend/src/services/patientService.js
import api from "./api";

export const getAllPatients = () => {
  return api.get("/patients");
};

export const getPatientById = (id) => {
  return api.get(`/patients/${id}`);
};

export const addPatient = (patient) => {
  return api.post("/patients", patient);
};

export const updatePatient = (id, patient) => {
  return api.put(`/patients/${id}`, patient);
};

export const deletePatient = (id) => {
  return api.delete(`/patients/${id}`);
};

File: hms-backend/src/main/java/com/hms/service/PatientService.java
package com.hms.service;

import java.util.List;
import com.hms.entity.Patient;

public interface PatientService {

    Patient savePatient(Patient patient);

    List<Patient> getAllPatients();

    Patient getPatientById(Long id);

    Patient updatePatient(Long id, Patient patient);

    void deletePatient(Long id);
}